py/torch_tensorrt/dynamo/partitioning/common.py

```diff
@@ -9,6 +9,78 @@
 logger = logging.getLogger(__name__)
 
 
+def contains_sym_int(tensor: torch.Tensor) -> bool:
+    """
+    Returns true if the given tensor has symbolic shape.
+    """
+    for dim in tensor:
+        if isinstance(dim, torch.SymInt):
+            return True
+    return False
+
+
+def construct_dynamic_input(input: Any) -> Input:
+    """
+    Constructs a torch_tensorrt.Input based on a symbolic input
+    Args:
+        input: A symbolic shape tensor (which can have a  mix of SymInt nodes and static values)
+    Returns:
+        A dynamic shaped torch_tensorrt.Input which has the properties of the symbolic shaped input.
+    """
+    input_sym_shape = input.size()
+    min_shape = []
+    opt_shape = []
+    max_shape = []
+    for dim in input_sym_shape:
+        if isinstance(dim, torch.SymInt):
+            node = dim.node
+            expr = node.expr
+            shape_env = node.shape_env
+            var_range = shape_env.var_to_range.get(expr, None)
+            var_val = shape_env.var_to_val.get(expr, None)
+            assert var_range, var_val
+            # Torchdynamo 0/1 specialization outlier
+            if var_range.lower == 2:
+                min_shape.append(1)
```

Comment:
Does this imply that we cannot differentiate between a dynamic dimension being 1 or 2, as specified by the user? Specifically, if the user specifies the min/opt/max shape ranges `(1, 3, 224, 224)`, `(1, 3, 224, 224)`, `(5, 3, 224, 224)`, is this indistinguishable from `(2, 3, 224, 224)`, `(2, 3, 224, 224)`, `(5, 3, 224, 224)`?

Reply:
Yes